src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -88,60 +84,102 @@ def process_breakdown_stables(data):
         "price",
     ]
 
-    metadata = {}
-    for key in must_have_metadata_fields:
-        metadata[key] = data[key]
-
-    for key in metadata_fields:
-        metadata[key] = data.get(key)
-
-    result = pl.DataFrame(rows, infer_schema_length=len(rows)).with_columns(
-        id=pl.lit(metadata["id"]),
-        name=pl.lit(metadata["name"]),
-        symbol=pl.lit(metadata["symbol"]),
+    metadata: Dict[str, Optional[str]] = {
+        key: data.get(key) for key in must_have_metadata_fields + metadata_fields
+    }
```

Comment:
I think that the list-comprehension is making the code shorter but obscuring the purpose